src/gsy_e/models/strategy/energy_parameters/energy_params_eb.py

```diff
@@ -21,37 +21,190 @@
 NOTE: The intraday product should not use this approach, since it already trades on single
 15-minutes timeslots.
 """
-from typing import Optional
+from abc import ABC, abstractmethod
+from typing import Optional, TYPE_CHECKING, Dict
 
 import pendulum
 from gsy_framework.enums import AvailableMarketTypes
 from gsy_framework.forward_markets.forward_profile import ForwardTradeProfileGenerator
 from gsy_framework.utils import convert_kW_to_kWh
 
+from gsy_e.constants import FLOATING_POINT_TOLERANCE
 from gsy_e.models.state import LoadState, PVState
 
+if TYPE_CHECKING:
+    from gsy_e.models.state import StateInterface
 
-class ConsumptionStandardProfileEnergyParameters:
+
+class ForwardEnergyParams(ABC):
+    """Common abstract base class for the energy parameters of the forward strategies."""
+    def __init__(self):
+        self._posted_energy_kWh: Dict = {}
+
+    def get_posted_energy_kWh(
+            self, market_slot: pendulum.DateTime, market_type: AvailableMarketTypes) -> float:
+        """Retrieve the already posted energy on this market slot."""
+        if market_type == AvailableMarketTypes.INTRADAY:
+            if market_slot not in self._posted_energy_kWh:
+                self._posted_energy_kWh[market_slot] = 0.
+
+            return self._posted_energy_kWh[market_slot]
+        if market_type == AvailableMarketTypes.DAY_FORWARD:
+            for slot in self.day_forward_slots(market_slot):
+                if slot not in self._posted_energy_kWh:
+                    self._posted_energy_kWh[slot] = 0.
```

Comment:
What do you think about using `defaultdict` for `self._posted_energy_kWh` and save this for loop here?
```suggestion
from collections import defaultdict
self._posted_energy_kWh: Dict =defaultdict(lambda: 0.0)
```

Reply:
Fixing this, thanks!